src/error.rs

```diff
@@ -40,6 +40,31 @@ impl fmt::Display for Error {
     }
 }
 
+impl Error {
+    /// Return true if the underlying error has the same type as T.
+    pub fn is<T: error::Error + 'static>(&self) -> bool {
+        let ie = self.inner_ref();
+        ie.is::<T>()
+    }
+
+    /// Return a reference to the lower level, inner error.
+    pub fn inner_ref(&self) -> &(error::Error + 'static) {
```

Comment:
This probably should be renamed `downcast_ref`.

Reply:
I'm fine with a rename, but the following prior art for `downcast_ref` seems quite different, signature wise:

https://doc.rust-lang.org/std/error/trait.Error.html#method.downcast_ref
https://doc.rust-lang.org/std/any/trait.Any.html#method.downcast_ref

Could alternatively name it `inner_error`?

io::Error used `get_ref` for this, would that be better?

https://doc.rust-lang.org/std/io/struct.Error.html#method.get_ref

Reply:
Yea, I wouldn't call it downcast unless we were actually downcasting inside.

Reply:
Ah sorry, you are right that given the current impl of `inner_ref`, it does not make sense to rename it to `downcast_ref`.

I wonder if there should be a `downcast_ref` fn that is generic over `T` to mirror the `is` function.

Reply:
The `into_inner` / `get_ref` / `get_mut` pattern could also possibly work. Thoughts @seanmonstar ?

Reply:
I don't have strong feelings either way. Probably not `into_inner`, unless there's a strong need for it...

Reply:
Ok, lets start w/ `get_ref` then.

Reply:
Renamed to `get_ref`